content/technotes/2020-02-26-tic-ghactions.Rmd

```diff
@@ -50,42 +50,86 @@ Furthermore {tic} aims to simplify the tedious process of setting up deployment
 ("Deployment" means that a CI run is allowed to perform a `git` push to the repository.)
 This is achieved by its helper packages such as {circle}, {ghactions} or {travis} which perform API calls to the respective CI providers.
 
+## GitHub Actions - the new kid on the block
 
+GitHub Actions is around since about one year.
+The hype after the launch was huge: People are used to expect properly developed products from GitHub and having an integrated CI solution without the overhead of connecting to a third party provider sounded promising.
 
-<!-- This is the markdown template for a blog post or tech note.  -->
-<!-- To generate your post from R Markdown, use that template instead. -->
+Indeed, GitHub Actions fulfilled this promise: It is way easier to get started with CI now: 
 
-<!-- Throughout this template, including the YAML,  -->
-<!-- you should change "post-template" to the slug of your post,  -->
-<!-- and "2019-06-04" to your publication date. -->
+- No extra account needs to be created
+- No restrictions on the available build platforms
+- No issues related to webhooks for third-party providers
+- CI settings configurable via the repository settings
 
-<!-- Save this file under /content/blog/YYYY-MM-DD-slug/index.md in the local copy of your roweb2 fork. -->
+However, R was again not among the languages with native support/examples, so it was not easy to get started.
+After a first shot from [Max Held]() via the [{ghactions}]() package, [Jim Hester]() sat down again and created an "action" which installs R on all platforms with the option to specify different R versions.
 
-<!-- ### Section heading in sentence case -->
+While this action is already usable, its still quite young and will mature over time.
+In {tic} we use [r-lib/actions]() as the base and apply some opinionated changes of which we think enhance the CI runs substantially.
+In the following we explain these changes to be as transparent as possible.
 
-<!-- Citation of the primary literature[^1].  -->
+### Caching
 
-<!-- Citation of a website[^2].  -->
+By default [r-lib/actions]() caches all direct dependencies of the package which is being tested on a successful run.
```

Comment:
How does this happen? I thought one needed an actual caching step?

Reply:
We have an official caching step - I'll link the one from our template.